### PR TITLE
Grid View for explorer 

### PR DESCRIPTION
**Pull request by mikestonecodes**
##### Description
Grid View for Contracts Explorer

##### Checklist
- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
UI

##### Refers/Fixes
Fixes: #275

### CODE REVIEW

**Comment by mikestonecodes** [comment]
Hey, it probably makes sense to do the token filters as a different issue. I'll just leave it not filtering in grid view for now so no time is wasted.  Pretty much completed everything else, but still have some smaller things to tackle.

**Comment by pelsasser** [comment]
@mikestonecodes - how is this coming? Was hoping to get this completed this week, but know you are busy.

**Comment by pelsasser** [comment]
Hey @mikestonecodes - sorry to keep pestering you, would you like us to take this off your plate so we can move it forward?  I don't want to be bothering you, but trying to move our milestone along on our end.

**Comment by mikestonecodes** [comment]
Hey I am going to finish this up right now , really sorry for the delay, been a bit crazy. should be able to give better timelines soon. thanks so much for being so understanding.

**Comment by mikestonecodes** [comment]
@pelsasser  @nitinrgupta  This is ready for review. sorry again for the delay

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "web3-fake-provider": "^0.1.0"
   },
   "dependencies": {
+    "@kennethormandy/react-fittext": "^0.3.0",
```
Comment (nitinrgupta) [inline package.json]: @mikestonecodes are we using this somewhere? If not, we can get rid of this dependency

File: src/components/ContractsList.js [hunk 2]
```diff
@@ -72,6 +94,57 @@ class ContractsList extends Component {
     this.setState({ [searchKey]: e.target.value });
   };
 
+  onGridChange = checked => {
+    console.log('efae', checked);
```
Comment (nitinrgupta) [inline src/components/ContractsList.js]: @mikestonecodes we can get rid of these console logs everywhere

File: src/less/ContractsList.less [hunk 3]
```diff
@@ -14,18 +14,180 @@
...
+    float: right;
+}
+.gridRow {
+    padding: 5px;
+    padding-left: 0px;
+    padding-right: 0px;
+    width: 214px;
+}
+.gridItem {
+    width: 255px;
```
Comment (nitinrgupta) [inline src/less/ContractsList.less]: I think defining specific width and height for the cards are causing responsiveness issues.

**Comment by nitinrgupta** [comment]
@mikestonecodes 

Here's the responsiveness issues I'm seeing
- This is at 1200px width
screen shot 2018-09-03 at 3 52 25 pm

- This is at 1400px width. Notice how the cards container are not in the center. Also the grid vs list switch option button position.
screen shot 2018-09-03 at 3 45 39 pm

I also saw one small issue
- If the contract is about to exprie soon, the top border of the card should go red.
Current behaviour:
screen shot 2018-09-03 at 3 52 43 pm

Expected behaviour:
screen shot 2018-09-03 at 4 05 51 pm


Also, the two icons are displaying wrong information right now. They should display the pair selected for the contract. Right now the left icon is having the exchange and the right icon is having the trading pair. 
@eswarasai @pelsasser will @mikestonecodes be able to parse which trading pair is chosen for the contract from the name?

**Comment by eswarasai** [comment]
@nitinrgupta -- Yes. The recent PR for standardizing the contract names, takes care of it. So the standard of the contract name is -- `REFERENCEASSET_COLLATERALSYMBOL_DATAPROVIDER_EXPIRATION_USERTEXT` in which you can get `REFERENCEASSET` and `COLLATERALSYMBOL` by passing the contract name to a `MARKET.js` utils function as per here -- 
https://github.com/MARKETProtocol/dApp/blob/f1448b3b2410f8cfc26686b04117030a6e0e01d1/src/components/DeployContract/Steps.js#L971

Note: Although currently the reference asset is in the format of `EOSETH` and not `EOS` like needed. I can put in a fix on `dApp:develop` which should take care of this.

cc: @mikestonecodes

**Comment by eswarasai** [comment]
@mikestonecodes -- Can you please rebase your branch with `upstream:develop`? We've just put in some fixes for some of the issues which are needed for wrapping up this PR. Let me know in case you need any help w.r.t. rebasing your branch. Looking forward to review and merge this PR. Thanks!

**Comment by mikestonecodes** [comment]
I completed all the feedback and rebased to current dApp:develop. Would be happy to help out more if needed.

**Comment by eswarasai** [comment]
@nitinrgupta -- Let me know once you've wrapped up with the design changes. I'll take a quick look at the code and do any further changes, if required. Thanks!

**Comment by pelsasser** [comment]
What is the status of this issue? are we actively working on this or is it iceboxed?

File: src/components/ContractsList.js [hunk 4]
```diff
@@ -91,6 +118,77 @@ class ContractsList extends Component {
...
+          Copy Oracle Query
+        </Button>
+      </div>
+    );
+
+    return (
+      <Popover
+        overlayClassName={'contractPopOver'}
+        content={popoverDetails}
+        placement={'bottomLeft'}
```
Comment (nitinrgupta) [inline src/components/ContractsList.js]: @eswarasai this should be `bottomRight`

Comment (eswarasai) [reply src/components/ContractsList.js]: @nitinrgupta -- You could've committed the change instead of commenting 🤦‍♂️

**Pull request merged**